extensions/publisher-cves/src/components/VulnerabilityChecker.vue

```diff
@@ -0,0 +1,272 @@
+<script setup lang="ts">
+import { useVulnsStore } from "../stores/vulns";
+import { usePackagesStore } from "../stores/packages";
+import { useContentStore } from "../stores/content";
+import type { Vulnerability } from "../stores/vulns";
+import type { Package } from "../stores/packages";
+import { computed } from "vue";
+
+// Import UI components
+import LoadingSpinner from "./ui/LoadingSpinner.vue";
+import StatusMessage from "./ui/StatusMessage.vue";
+
+// Import vulnerability components
+import StatsPanel from "./vulnerability/StatsPanel.vue";
+import EmptyState from "./vulnerability/EmptyState.vue";
+import VulnerabilityList from "./vulnerability/VulnerabilityList.vue";
+import ArrowTopRight from "./icons/ArrowTopRight.vue";
+
+interface VulnerablePackageItem {
+  packageInfo: Package;
+  vulnerabilities: Vulnerability[];
+  repo: "pypi" | "cran";
+  latestFixedVersion: string | null;
+}
+
+const vulnStore = useVulnsStore();
+const packagesStore = usePackagesStore();
+const contentStore = useContentStore();
+
+const packages = computed(() => {
+  // Get the current content's packages
+  const currentId = contentStore.currentContentId;
+  if (!currentId) return [];
+
+  const contentItem = packagesStore.contentItems[currentId];
+  return contentItem ? contentItem.packages : [];
+});
+
+// Track loading states
+const isLoadingVulns = computed(() => vulnStore.isLoading);
+const isLoadingPackages = computed(() => {
+  // Check if the current content is loading
+  const currentId = contentStore.currentContentId;
+  if (!currentId) return false;
+
+  const contentItem = packagesStore.contentItems[currentId];
+  return contentItem ? contentItem.isLoading : false;
+});
+const isLoading = computed(
+  () => isLoadingPackages.value || isLoadingVulns.value,
+);
+const hasError = computed(() => {
+  const currentId = contentStore.currentContentId;
+  if (!currentId) return vulnStore.error;
+
+  const contentItem = packagesStore.contentItems[currentId];
+  return vulnStore.error || (contentItem ? contentItem.error : null);
+});
+
+// Check if we have packages to analyze
+const hasPackages = computed(() => {
+  return packages.value.length > 0;
+});
+
+// Extract the fixed version from the vulnerability ranges data
+function getFixedVersion(vuln: Vulnerability): string | null {
+  if (!vuln.ranges || !Array.isArray(vuln.ranges) || vuln.ranges.length === 0) {
+    return null;
+  }
+
+  // Look through all ranges
+  for (const range of vuln.ranges) {
+    // Look for events with a "fixed" property
+    if (range.events && Array.isArray(range.events)) {
+      for (const event of range.events) {
+        if (event.fixed) {
+          return event.fixed;
+        }
+      }
+    }
+  }
+
+  return null;
+}
+
+// Go back to content list
+function goBack() {
+  contentStore.currentContentId = undefined;
+}
+
+// Find vulnerable packages by comparing package data with vulnerability data
+const vulnerablePackages = computed<VulnerablePackageItem[]>(() => {
```

Comment:
This collection of vulnerable packages is more complex than I'd like it to be. I figured I could create a follow-up to come back to this, move some of the logic into the store, and refactor it a bit.

I'm happy to address feedback on this PR, but it felt like a good follow-up to me as I was digging into it.